interp/constrintern.ml

```diff
@@ -2043,6 +2055,8 @@ let internalize globalenv env pattern_mode (_, ntnvars as lvar) c =
         intern_notation intern env ntnvars loc ntn args
     | CGeneralization (b,a,c) ->
         intern_generalization intern env ntnvars loc b a c
+    | CPrim _ when !disable_elaboration ->
+       user_err ?loc (Pp.strbrk "Primitive notations are disabled when elaboration is blocked.")
```

Comment:
How are you supposed to use primitive ints/floats then?

Reply:
Indeed, that was something that I was wondering too. Maybe I should remove this.

Reply:
What is the "kernel" story about floats in the gallina chapter? There is no discriminating syntax for floats in the surface language, right? Or, do you intend to only accept `XX%float` in the gallina chapter. In the latter case, there can be a special case here to let `CDelimiters` of `CPrim` goes through when the key is "float"?

Reply:
Actually, `Check 4%float` does not work in `-nois` mode. Should we conclude that floating-point numbers are a "library" thing, or that a syntax should be natively provided?

Reply:
My own conclusion is that it is reasonable to consider floating-point and int63 arithmetic as only a library thing. The kernel only contains the "pre-guarantee" that some concrete user-level datatypes and operations on them can have their kernel representation and kernel computation rules "optimized". It does not give names to them.

The "strange" thing is that the optimized representation of the elements of the data-types are in advance in the structure of terms (`Float` and `Int` fields of `constr`) but the operations on these are not in advance in the environment (the `Primitive`-style form of `Const`). This is because only the optimization of reduction rules for them matters, not the representation itself.

Reply:
I perfectly agree with that conclusion. In particular, parsers for primitive integers and floats are implemented as plugins (in `plugins/syntax`) hence loaded only with the libraries. Similarly, the primitive functions that could be used to craft primitive integers/floats are only linked (with the `Primitive` vernacular) in the libraries. Thus, it is impossible to get a primitive value without loading the related library.

In fact, this is quite moral. Ideally, these would be purely library things but for performance, some computations need to sneak into the kernel. But it seems preferable to put as few things as possible inside the kernel.